rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -182,13 +182,13 @@ rcl_logging_ret_t rcl_logging_external_initialize(
       name_buffer, sizeof(name_buffer),
       "%s/%s_%i_%" PRId64 ".log", logdir,
       basec, rcutils_get_pid(), ms_since_epoch);
+    if (!file_name_provided) {
```

Comment:
It can work, but it seems not a good practice as what if there is some extra logic that can return between `basec` allocation and `rcutils_snprintf` in the future.

Reply:
yeah i thought that too.

the things are,
- we cannot use if statement with `RCPPUTILS_SCOPE_EXIT`, since it will deallocate before used.
- besides, i do want to avoid unnecessary memory allocation when we can use `file_name_prefix`.

there are some other places to do deallocation like this, and it is really close not to miss in this case.

Reply:
```cpp
    RCPPUTILS_SCOPE_EXIT(
    {
      if (!file_name_provided) {
        allocator.deallocate(basec, allocator.state);
      }
    });
```

> we cannot use if statement with RCPPUTILS_SCOPE_EXIT, since it will deallocate before used.

I am not sure which way you used, is it like below,

```cpp
    if (!file_name_provided) {
      RCPPUTILS_SCOPE_EXIT(
      {
        allocator.deallocate(basec, allocator.state);
      });
    }
```

> there are some other places to do deallocation like this.

They're not a good practice as well.